src/mahoji/lib/abstracted_commands/camdozaalCommand.ts

```diff
@@ -0,0 +1,126 @@
+import { Time } from 'e';
+import { Bank } from 'oldschooljs';
+import { SkillsEnum } from 'oldschooljs/dist/constants';
+
+import { ActivityTaskOptionsWithQuantity } from '../../../lib/types/minions';
+import { formatDuration } from '../../../lib/util';
+import addSubTaskToActivityTask from '../../../lib/util/addSubTaskToActivityTask';
+import { calcMaxTripLength } from '../../../lib/util/calcMaxTripLength';
+
+async function miningCommand(user: MUser, channelID: string, quantity: number | undefined) {
+	if (user.skillLevel(SkillsEnum.Mining) < 14) {
+		return 'You need at least level 14 Mining to fish in the Ruins of Camdozaal.';
+	}
+
+	const maxTripLength = calcMaxTripLength(user, 'CamdozaalMining');
+	const timePerMine = 4.5 * Time.Second;
+	if (!quantity) {
+		quantity = Math.floor(maxTripLength / timePerMine);
+	}
+	const duration = timePerMine * quantity;
+
+	if (duration > maxTripLength) {
+		return `${user.minionName} can't go on trips longer than ${formatDuration(
+			maxTripLength
+		)}, try a lower quantity. The highest amount of Barronite rocks you can mine is ${Math.floor(
+			maxTripLength / timePerMine
+		)}.`;
+	}
+
+	await addSubTaskToActivityTask<ActivityTaskOptionsWithQuantity>({
+		userID: user.id,
+		channelID: channelID.toString(),
+		quantity,
+		duration,
+		type: 'CamdozaalMining'
+	});
+
+	return `${user.minionName} is now mining in the Ruins of Camdozaal, it will take around ${formatDuration(
+		duration
+	)} to finish.`;
+}
+
+async function smithingCommand(user: MUser, channelID: string, quantity: number | undefined) {
+	if (user.skillLevel(SkillsEnum.Smithing) < 14) {
+		return 'You need at least level 14 Smithing to smith in the Ruins of Camdozaal';
+	}
+
+	const maxTripLength = calcMaxTripLength(user, 'CamdozaalSmithing');
+	const timePerSmith = 3.5 * Time.Second;
+	if (!quantity) {
+		quantity = Math.floor(maxTripLength / timePerSmith);
+	}
+	const duration = timePerSmith * quantity;
+
+	if (user.bank.amount('Barronite deposit') < quantity) {
+		return "You don't have enough Barronite desposit's to smelt.";
```

Comment:
Should add something like a maxCanDo and default it to that rather than just erroring, see the mixCommand/smeltCommand for a working example

Reply:
I attempted to do something like the maxCanDo but couldn't really figure it out and went with this. Whats the goal for how the end result look like?

Reply:
I was able to add something similar and I believe it gives the same results as if you were smiting bars.